src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlEnums.cs

```diff
@@ -124,6 +124,7 @@ private static bool _IsCharType(SqlDbType type) =>
             type == SqlDbType.Char ||
             type == SqlDbType.VarChar ||
             type == SqlDbType.Text ||
+            type == SqlDbTypeExtensions.Json ||
```

Comment:
Is the usual read operation working as expected after adding this? (We had a problem earlier in read which was fixed by removing this)

Reply:
Yes. The read issue was caused due to the existing code path trying to find the encoding and collation for char datatypes. The fix for the same is already in place for read/write changes.